compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1397,8 +1397,11 @@ class JSCodeGen()(using genCtx: Context) {
     } else /*if (isScalaJSDefinedJSClass(currentClassSym)) {
       genJSSuperCall(tree, isStat)
     } else*/ {
+      /* #3013 `qual` can be `this.$outer()` in some cases since Scala 2.12,
+       * so we call `genExpr(qual)`, not just `genThis()`.
+       */
       val superCall = genApplyMethodStatically(
-          genThis()(sup.span), sym, genActualArgs(sym, args))
+          genExpr(qual), sym, genActualArgs(sym, args))
```

Comment:
@sjrd The change causes non-termination in existing tests (not `RegressionTest.scala`). Any ideas?

Reply:
Hum. If I apply the same change on `master`, I can successfully run all existing tests (but not `RegressionTest.scala`). So it seems that it is related to other changes in this PR. :s

Reply:
Perhaps this has to do with the fact that the JVM back-end uses `DesugaredSelect` instead of `Select`. `DesugaredSelect` is a helper in `DottyBackendInterface` that turns an `Ident` whose `tpe` is a prefix type into a `Select`.

If this becomes too complicated, drop the Scala.js part, and I'll have a look once this PR is merged.

Reply:
Now it works after rebase, without using `DesugaredSelect`. The `RegressionTest.scala` test causes non-termination, so I didn't enable it.

Reply:
Cool thank you :)